docs/workforce-management-framework/topics/con-security-introduction.adoc

```diff
@@ -0,0 +1,61 @@
+[]
+= Introduction to {WFM-RC-NameShort} security features
+
+{WFM-RC-NameShort} cloud server gives developers a choice one of the following authentication and authorisation providers
+link:http://www.keycloak.org/[Keycloak] or link:http://passportjs.org/[Passport.js].
+{WFM-RC-NameShort} is build around a security interface, which abstracts the developer away from any underlying
+authentication and authorisation libraries.
+Both implementations can be swapped at any time without breaking internal API of the {WFM-RC-NameShort} modules and reference applications.
+Users can also provide their own security implementation by implementing their own security interface.
+
+Other security measures are incorporated into our reference applications.
+Please note that all link:https://expressjs.com/[express] based routes outside WFM modules will also need to use these security interfaces.
+
+== How to Apply Access Control
+The Security modules interface can be implemented with both Keycloak and Passport.js.
+You need to import EndpointSecurity interface. Apply either
+link:https://github.com/feedhenry-raincatcher/raincatcher-core/blob/{WFM-RC-Branch}/demo/server/src/modules/keycloak/index.ts[Keycloak]
+implementation initialisation or link:https://github.com/feedhenry-raincatcher/raincatcher-core/blob/{WFM-RC-Branch}/demo/server/src/modules/passport-auth/index.ts[Passport.js]
+to the interface as show in our link:https://github.com/feedhenry-raincatcher/raincatcher-core/blob/{WFM-RC-Branch}/demo/server/src/modules/index.ts[cloud server].
+You can then use this security interface protect method to guard routes with roles.
+
+== Passport.js
+Passport.js is a lightweight authentication middleware for Node.js.
+Passport.js offers comprehensive set of link:http://passportjs.org/docs/configure[strategies] that support authentication
+using a username and password, Facebook, Twitter, and more.
+For more information see link:http://passportjs.org/[passport.org].
+
+- Can be bundled into existing node.js server
+- Available by default when running our reference server application
+- Covers authentication of express routes
```

Comment:
"Covers authentication of express routes" > "Covers authentication `and access control` of express routes"